typedb/stream/response_part_iterator.py

```diff
@@ -74,9 +74,7 @@ def _has_next(self) -> bool:
             raise TypeDBClientException.of(ILLEGAL_STATE)
 
     def __next__(self) -> transaction_proto.Transaction.ResPart:
-        if self._bidirectional_stream.get_error() is not None:
```

Comment:
the error will come from the backing `Queue`, rather than the bidirectional stream.